src/input/type_record.cc

```diff
@@ -180,7 +181,7 @@ Record &Record::derive_from(AbstractRecord &parent) {
 	data_->parent_ptr_.push_back( boost::make_shared<AbstractRecord>(parent) );
 
 	if (data_->keys.size() == 0) {
-    	data_->declare_key("TYPE", boost::shared_ptr<TypeBase>(NULL), Default::obligatory(), "Sub-record Selection.");
+    	data_->declare_key("TYPE", boost::shared_ptr<TypeBase>(NULL), Default( type_name() ), "Sub-record Selection.");
```

Comment:
- set of Default is moved from Record::finish
- abstract records haven't keys, record is registered into abstract record in add_child method what is called during Record::close()
- declare_type_key set Default::obligatory